src/Compilers/CSharp/Portable/Symbols/Synthesized/InjectedAttributeSymbol.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Injected attribute symbols are injected early in the compilation, and so can be referenced in source.
+    /// But we track their usage and only emit them if they are used.
+    /// Their method bodies are always compiled, in case we do need to emit them.
+    /// </summary>
+    internal abstract class InjectedAttributeSymbol : SynthesizedEmbeddedAttributeSymbolBase
+    {
+        // All the diagnostics involved in constructing this symbol will only be produced
+        // if the symbol is referenced and so ends up emitted.
+        // We collect all those diagnostics here.
+        internal readonly DiagnosticBag Diagnostics;
```

Comment:
>internal readonly DiagnosticBag Diagnostics; [](start = 8, length = 44)

Consider avoiding exposing this bag to external consumer. For example add an API GetDiagnostics that passes a bag to add diagnostics into. This is a private storage that no one has access to.